lib/logging/lib/logging.c

```diff
@@ -140,23 +143,31 @@ log_impl(
     *timer = now + log_squelch_ns;
 
     if (err) {
+        int16_t ctx;
         char buf[256];
 
         merr_strerror(err, buf, sizeof(buf));
 
-        rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d %s: %s: %s (%d)\n",
-            file, lineno, func, fmt, buf, merr_errno(err));
+        ctx = merr_ctx(err);
+        if (ctx == 0 || !log_err_ctx_stringify) {
+            rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d: %s: %s: %s (%d)\n",
+                file, lineno, func, fmt, buf, merr_errno(err));
+        } else {
+            rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls),
+                "[HSE] %s:%d: %s: %s: %s (%d): %s (%d)\n",
+                file, lineno, func, fmt, buf, merr_errno(err), log_err_ctx_stringify(ctx), ctx);
+        }
     } else {
-        rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d %s: %s\n",
+        rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d: %s: %s\n",
             file, lineno, func, fmt);
     }
     if (rc >= sizeof(log_buffer_tls)) {
-        backstop(LOG_ERR, "[HSE] %s:%d %s: scatch buffer size too small, needed %d for %s:%d\n",
+        backstop(LOG_ERR, "[HSE] %s:%d: %s: scatch buffer size too small, needed %d for %s:%d\n",
```

Comment:
You really need to print the message buffer, even if it's truncated.

Reply:
Greg wanna check the logic here again to see if it fixes your concern?

Reply:
Doesn't appear that anything has changed?  Most system like this just put a ". . ." at the end of any message that overflows the output buffer, waaaaay better to get truncated output than nothing at all...

Reply:
Greg, notice that the early return is gone, so it'll log this message and then a portion of the actual message.